File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPartsAndProductsAllocated/EngineerVisitPartProductAllocatedValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitPartProductAllocateds
    {
        // make sure that contact object is valid
        public class EngineerVisitPartProductAllocatedValidator : BaseValidator
        {
            public void Validate(EngineerVisitPartProductAllocated oEngineerVisitPartAllocated)
            {

                // make sure that contact object is valid
                if (oEngineerVisitPartAllocated.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oEngineerVisitPartAllocated.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oEngineerVisitPartAllocated.Type.Trim().Length == 0)
                {
                    AddCriticalMessage("Item Type Missing");
                }

                if (oEngineerVisitPartAllocated.PartProductID == 0)
                {
                    AddCriticalMessage("Item Missing");
                }

                if (!(oEngineerVisitPartAllocated.Quantity > 0))
                {
                    AddCriticalMessage("Quantity Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPartsAndProductsAllocated/EngineerVisitPartProductAllocatedValidator.vb
Namespace Entity

    Namespace EngineerVisitPartProductAllocateds

        Public Class EngineerVisitPartProductAllocatedValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerVisitPartAllocated As EngineerVisitPartProductAllocated)

                'make sure that contact object is valid
                If oEngineerVisitPartAllocated.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerVisitPartAllocated.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oEngineerVisitPartAllocated.Type.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Item Type Missing")
                End If
                If oEngineerVisitPartAllocated.PartProductID = 0 Then
                    Me.AddCriticalMessage("Item Missing")
                End If
                If Not oEngineerVisitPartAllocated.Quantity > 0 Then
                    Me.AddCriticalMessage("Quantity Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace
